packages/toolpad-app/src/appDom/index.ts

```diff
@@ -727,6 +727,16 @@ export function moveNode<Parent extends AppDomNode, Child extends AppDomNode>(
   return setNodeParent(dom, node, parent.id, parentProp, parentIndex);
 }
 
+export function duplicateNode<Child extends ElementNode>(dom: AppDom, node: Child) {
+  if (!node.parentId || !node.parentProp || !node.parentIndex) {
+    throw new Error(`Node: "${node.id}" can't be duplicated`);
+  }
+
+  const newNode = createElement(dom, node.attributes.component!.value, node.props, node.layout);
+
+  return setNodeParent(dom, newNode, node.parentId, node.parentProp, node.parentIndex);
```

Comment:
The parentIndex needs to be different. 
Parent indexes identify the ordered position of a component within a container, so because we want to add the new component **after** the old one, we need to make the parent index be between the old component and the component next to it - or if there is no component next to it, get the last parent index.
Probably the function `getNewParentIndexAfterNode` is all we need to use here (it does all that for us).

Reply:
I believe we can use the `addNode` function to do that for us?

Reply:
I've just tried this and I'm getting TS error which complains that the prop I'm passing doesn't belong neither to child nor to parent:
<img width="1049" alt="image" src="https://user-images.githubusercontent.com/437214/196407824-98ef1ce5-5633-49b3-9016-5e1ee45ba24f.png">

Not really sure how to fix this, any tips?